outbound.go

```diff
@@ -134,9 +134,17 @@ func (c *Connection) beginCall(ctx context.Context, serviceName, methodName stri
 	if err := call.writeMethod([]byte(methodName)); err != nil {
 		return nil, err
 	}
+
 	return call, nil
 }
 
+func (c *Connection) outboundCtxCancel() {
+	// outbound contexts are created by callers, can't cancel them.
+	// However, we shouldn't be trying to cancel them, so log.
+	c.log.Warn("unexpected cancel of outbound context")
```

Comment:
worried that this might be overly verbose in practice - should it be a debug log instead?

Reply:
Can change the level, though this should never happen. This is passed to the message exchange's `ctxCancel` func, and that's only used when a cancel frame is received. A cancel frame should only be received for inbound calls, which will cancel the real context. This is basically a no-op used for outbound message exchanges which aren't expected to receive a cancel.

None of the tests trigger this, and so this shouldn't happen unless a bad TChannel implementation sends an unexpected frame.

Reply:
> ...and so this shouldn't happen unless a bad TChannel implementation sends an unexpected frame.

yea, mostly worried that if this is somehow triggered by an edge case there won't be a way to silence it unless we roll back. upvote for changing it to debug level.